src/lib.rs

```diff
@@ -178,6 +178,23 @@ pub struct Peripheral {
 }
 
 impl Peripheral {
+    pub fn derive_from(&self, other: &Peripheral) -> Peripheral {
+        let mut derived = self.clone();
+        if derived.group_name.is_none() && other.group_name.is_some() {
```

Comment:
minor nit: the conditionals here don't need to check `other.XXX`.   If they're `is_none` then the `derived` value will get assigned to `None` anyway.  It just feels a little less repetitive:

```
  if derived.group_name.is_none() {
       derived.group_name = other.group_name.clone();
  }
```

Reply:
I second this. I'll merge this when this has been fixed.